locallib.php

```diff
@@ -3660,5 +3680,41 @@ function bigbluebuttonbn_create_meeting_metadata(&$bbbsession) {
     if ((boolean) \mod_bigbluebuttonbn\locallib\config::get('meetingevents_enabled')) {
         $metadata['analytics-callback-url'] = $bbbsession['meetingEventsURL'];
     }
+    // Special metadata for Opencast recordings (passing opencast seriesid of the course as opencast-dc-isPartOf as metadata).
+    if ((boolean) \mod_bigbluebuttonbn\locallib\config::get('oc_recording')
+        && bigbluebuttonbn_check_opencast($bbbsession['course']->id)) {
+        $metadata['opencast-dc-isPartOf'] = bigbluebuttonbn_check_opencast($bbbsession['course']->id);
+    }
```

Comment:
You are fetching the Opencast series ID twice which seems to be an overhead.
How about something like this:
```suggestion
    if ((boolean) \mod_bigbluebuttonbn\locallib\config::get('oc_recording') {
        $ocseriesid = bigbluebuttonbn_check_opencast($bbbsession['course']->id);
        if ($ocseriesid != false) {
            $metadata['opencast-dc-isPartOf'] = $ocseriesid;
        }
    }
```